src/server/svr_jobfunc.c

```diff
@@ -4705,11 +4705,10 @@ start_end_dur_wall(void *pobj, int objtype)
 
 		case  7:	/*start, end, duration*/
 			if ((pstime->at_val.at_long < time_now) ||
-				(petime->at_val.at_long < pstime->at_val.at_long))
-				rc = -1;
-			else if ((pduration->at_val.at_long <= 0) ||
-				((petime->at_val.at_long - pstime->at_val.at_long)<
-				pduration->at_val.at_long))
+				(petime->at_val.at_long < pstime->at_val.at_long) ||
+				(pduration->at_val.at_long <= 0) ||
+				((petime->at_val.at_long - pstime->at_val.at_long) <
```

Comment:
Shouldn't this be '!=' instead of '<'?  If you specify all three (start, end, and duration), then the duration better equal end-start.

Reply:
I don't really know why it's `<` instead of `!=`. The only change I made here was to get rid of redundant else ifs.

I'll change it and see how it affects the code.